src/Api/Clients/ProductClient.cs

```diff
@@ -10,39 +12,94 @@ public interface IProductClient
         Task<ApiResponse<IEnumerable<Product>>> GetProductsAsync();
         Task<ApiResponse<ProductTicker>> GetProductTickerAsync(string productId);
         Task<ApiResponse<OrderBook>> GetOrderBookAsync(string productId, int level = 1);
-    }
+		Task<List<Candle>> GetHistoricRatesAsync(string i_product_id, DateTimeOffset i_start, DateTimeOffset i_end, TimeSpan i_granularity);
+	}
```

Comment:
Indenting again, you should probably have a look at your IDE/code editor settings with regards to tabs & spaces